app/assets/javascripts/discourse/app/routes/unknown.js

```diff
@@ -11,7 +11,13 @@ export default DiscourseRoute.extend({
       if (results.found) {
         // Avoid polluting the history stack for external links
         transition.abort();
-        DiscourseURL.routeTo(results.target_url);
+
+        let url = results.target_url;
+        if (transition._discourse_anchor) {
+          url += `#${transition._discourse_anchor}`;
```

Comment:
Is it possible there's already an anchor and this would be adding a second one?

Reply:
It is! I've updated the code. It now replaces the anchor from the permalink when present.